pallets/aleph/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+//! This pallet is a runtime companion of Aleph finality gadget.
+//!
+//! Currently, it only provides support for changing sessions but in the future
+//! it will allow reporting equivocation in AlephBFT.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+mod migrations;
+
+use sp_std::prelude::*;
+
+use frame_support::{
+    sp_runtime::BoundToRuntimeAppPublic,
+    traits::{OneSessionHandler, StorageVersion},
+    Parameter,
+};
+pub use pallet::*;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(2);
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::{pallet_prelude::*, sp_runtime::RuntimeAppPublic};
+    use frame_system::pallet_prelude::BlockNumberFor;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type AuthorityId: Member
+            + Parameter
+            + RuntimeAppPublic
+            + Default
+            + MaybeSerializeDeserialize;
+    }
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_runtime_upgrade() -> frame_support::weights::Weight {
+            migrations::v0_to_v1::migrate::<T, Self>() + migrations::v1_to_v2::migrate::<T, Self>()
+        }
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn authorities)]
+    pub(super) type Authorities<T: Config> = StorageValue<_, Vec<T::AuthorityId>, ValueQuery>;
+
+    impl<T: Config> Pallet<T> {
```

Comment:
do I understand correctly, that the newest version of this pallet is only to serve `authorities` set internally to the AlephBFT (which itself is computed elsewhere)? if so, then it seems as a massive overhead for something that seems achievable without dedicated pallet

Reply:
Talked through :)

Reply:
This is the way.